src/main/webapp/app/text-assessment/text-assessment.component.ts

```diff
@@ -290,14 +292,33 @@ export class TextAssessmentComponent implements OnInit, OnDestroy, AfterViewInit
         this.formattedSampleSolution = this.artemisMarkdown.htmlForMarkdown(this.exercise.sampleSolution);
 
         this.result = this.participation.results[0];
-        this.hasComplaint = this.result.hasComplaint;
+        if (this.result.hasComplaint) {
+            this.getComplaint();
+        }
 
         this.loadFeedbacks(this.result.feedbacks || []);
         this.busy = false;
         this.validateAssessment();
         this.checkPermissions();
     }
 
+    getComplaint(): void {
+        this.complaintService.findByResultId(this.result.id).subscribe(
```

Comment:
Also here a loading indicator would be great.
If you don't want to repeat yourself, it might be a good idea to wrap the complaint segment into anothe component.

Reply:
There should also be a third state: No complaints.